### PR TITLE
Consolidate and clean up operator controllers

### PR DESCRIPTION
**Pull request by jim-minter**

### CODE REVIEW

**Comment by jim-minter** [comment]
@troy0820 as discussed, ptal

File: pkg/operator/controllers/routefix/routefix_controller.go [hunk 1]
```diff
@@ -73,49 +69,22 @@ func (r *RouteFixReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error
...
-		return dynamichelper.CreateOrder(uns[i], uns[j])
-	})
-
-	for _, un := range uns {
-		err = dh.Ensure(ctx, un)
-		if err != nil {
-			r.log.Error(err)
-			return reconcile.Result{}, err
-		}
+	err = dh.Ensure(ctx, uns...)
```
Comment (nilsanderselde) [inline pkg/operator/controllers/routefix/routefix_controller.go:84]: nice variadic function

**Review by nilsanderselde — APPROVED** [review]
LGTM

File: pkg/operator/controllers/genevalogging/genevalogging.go [hunk 2]
```diff
@@ -68,7 +40,12 @@ func (g *genevaLogging) securityContextConstraints(ctx context.Context, name, se
 	return scc, nil
 }
 
-func (g *genevaLogging) daemonset(r azure.Resource) *appsv1.DaemonSet {
+func (g *GenevaloggingReconciler) daemonset(cluster *arov1alpha1.Cluster) (*appsv1.DaemonSet, error) {
+	r, err := azure.ParseResourceID(cluster.Spec.ResourceID)
+	if err != nil {
+		return nil, err
+	}
```
Comment (troy0820) [inline pkg/operator/controllers/genevalogging/genevalogging.go:47]: Old function used to use `r` to create daemonset, now we are using the `*arov1alpha1.Cluster` instead.  Is this necessary?

Comment (jim-minter) [reply pkg/operator/controllers/genevalogging/genevalogging.go:47]: Yes.

**Pull request merged**